### PR TITLE
Allow changing of Plane modes using AuxSwitches

### PR DESCRIPTION
**Pull request by peterbarker**
Rover and Copter already have support for doing this, and someone recently asked for it.

Note that we haven't integrated the mode switch into RC_Channel in Plane yet; it still does checking of switch position, debouncing etc, which Rover and Copter leave to the base RC_Channel object.  Thus the call to `plane.reset_control_switch()`

### CODE REVIEW

**Comment by peterbarker** [comment]
Closes #8020

**Comment by peterbarker** [comment]
This has been flight-tested now.

File: ArduPlane/RC_Channel.cpp [hunk 1]
```diff
@@ -27,13 +27,40 @@ bool RC_Channels_Plane::has_valid_input() const
...
+void RC_Channel_Plane::do_aux_function_change_mode(const Mode::Number number,
+                                                   const aux_switch_pos_t ch_flag)
+{
+    switch(ch_flag) {
+    case HIGH: {
+        // engage mode (if not possible we remain in current flight mode)
+        const bool success = plane.set_mode_by_number(number, MODE_REASON_TX_COMMAND);
+        if (plane.control_mode != &plane.mode_initializing) {
+            if (success) {
+                AP_Notify::events.user_mode_change = 1;
```
Comment (WickedShell) [inline ArduPlane/RC_Channel.cpp]: Not new to this, but whatever clears this flag? Presumably it gets cleared somewhere, but I can't figure out where/how....

Comment (peterbarker) [reply ArduPlane/RC_Channel.cpp]: > Not new to this, but whatever clears this flag? Presumably it gets cleared somewhere, but I can't figure out where/how....

https://github.com/ardupilot/ardupilot/blob/master/libraries/AP_Notify/AP_Notify.cpp#L330

**Comment by Jaaaky** [comment]
@peterbarker Can you please explain what is the point here?
Why not flt_modech is not enough? It's already working on rc switch.

**Comment by peterbarker** [comment]
@Jaaaky It's a long-standing feature of Copter and slightly-less-longstanding feature of Rover.  We should always try to make our vehicles work more like the others where it makes sense to.

See the referenced issue (#8020) for one user's use case.

**Pull request merged**